addons/web/static/src/core/macro.js

```diff
@@ -75,6 +76,27 @@ class Macro {
         this.onComplete = this.onComplete || (() => {});
         this.onStep = this.onStep || (() => {});
         this.stepElFound = new Array(this.steps.length).fill(false);
+        this.stepHasStarted = new Array(this.steps.length).fill(false);
+    }
+
+    async start(target = document) {
+        this.observer = new MacroMutationObserver(() => this.debounceAdvance("mutation"));
+        this.observer.observe(target);
+        this.debounceAdvance("next");
+    }
+
+    get debounceDelay() {
+        let delay = Math.max(this.checkDelay ?? 750, 50);
+        // Called only once per step.
+        if (!this.stepHasStarted[this.currentIndex]) {
+            delay = this.currentIndex === 0 ? 0 : 50;
+            this.stepHasStarted[this.currentIndex] = true;
```

Comment:
A getter should never set anything as a side-effect, change this to a regular function instead